Doc/whatsnew/3.8.rst

```diff
@@ -750,6 +750,11 @@ Deprecated
   module is deprecated in favor of the :func:`inspect.signature`
   API.  (Contributed by Pablo Galindo in :issue:`36751`.)
 
+* Using :data:`NotImplemented` in a boolean context has been deprecated,
+  as it is almost exclusively the result of incorrect rich comparator
+  implementations. It will be made a :exc:`TypeError` in Python 4.0.
```

Comment:
Maybe "some future version" rather than "Python 4.0"

Reply:
Sure. I got the impression most deprecation announcements tried to pin down a specific version (and I assumed whoever merged the patch would tweak the specific version if so decided), but I've got no objection to leaving it open.

Reply:
Pushed change in 33a1802, changed wording from "in Python 4.0" to "in a future version of Python"